src/transformState.js

```diff
@@ -5,7 +5,21 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(state, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const removed of action.keysToRemove) {
+        delete state[removed];
+      }
+    } else if (action.type === 'clear') {
+      for (const prop in state) {
+        delete state[prop];
+      }
+    }
+  }
+
+  return state;
```

Comment:
You don't need to return the `state` object at the end of the function since the changes are made directly to the original object.